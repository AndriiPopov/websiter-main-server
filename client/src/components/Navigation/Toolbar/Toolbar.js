import React from 'react';
import { withRouter } from 'react-router-dom';

import classes from './Toolbar.module.css';
import Logo from '../../Logo/Logo';
import DrawerToggle from '../SideDrawer/DrawerToggle/DrawerToggle';
import NavigationItems from '../../SiteBuilderNavigation/Toolbar/NavigationItems/NavigationItems';

const toolbar = props => (
    <header className={classes.Toolbar}>
        <DrawerToggle clicked={props.drawerClicked} />
        <div className={classes.Logo}>
            <Logo />
        </div>
        { props.location.pathname === '/editor' ?
            <div className={classes.BuilderMenu}>
                <NavigationItems />
            </div>
            : null
        }
    </header>
);

export default withRouter(toolbar);