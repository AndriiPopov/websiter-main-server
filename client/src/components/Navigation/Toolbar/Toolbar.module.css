.Toolbar {
    height: 56px;
    width: 100%;
    position: fixed;
    top: 0px;
    left: 0px;
    border: 1px solid rgb(232, 232, 232);
    display: flex;
    align-items: center;
    padding: 0px 20px;
    box-sizing: border-box;
    z-index: 90;
}

.Toolbar nav {
    height: 100%;
    
}

.Logo {
    margin-left: 50px;
    height: 80%;
}

.BuilderMenu {
    flex: 1;
    height: 100%;
    display: flex;
    align-items: flex-end;
    padding-left: 50px;
}