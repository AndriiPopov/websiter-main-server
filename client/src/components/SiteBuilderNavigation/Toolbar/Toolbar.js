import React from 'react';
import { connect } from 'react-redux';

import classes from './Toolbar.module.css';
import ToolSet from './ToolsSet/ToolsSet';
import Sections from './ToolsSet/TopMenuSections/Sections';
import Boxes from './ToolsSet/TopMenuSections/Boxes';
import Text from './ToolsSet/TopMenuSections/Text';

export const toolbar = props => (
    <header className={classes.Toolbar}>
        { !props.previewIsOn ?
                <div className={classes.ToolSetsContainer}>
                    <ToolSet title="sections" >
                        <Sections />
                    </ToolSet>
                    <ToolSet title="boxes" >
                        <Boxes />
                    </ToolSet>
                    <ToolSet 
                        title="text"
                        textEditor={props.textEditor} >
                        <Text />
                    </ToolSet>
                </div>
            : null
        }
    </header>
);

const mapStateToProps = (state, props) => {
    return {
        previewIsOn: state.builder.present.preview        
    };
};

export default connect(mapStateToProps)(toolbar);