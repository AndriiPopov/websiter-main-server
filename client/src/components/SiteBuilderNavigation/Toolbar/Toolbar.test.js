import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { toolbar as Toolbar } from './Toolbar';

configure({adapter: new Adapter()});

describe('Toolbar', () => {
    it('should be defined', () => {
        expect(Toolbar).toBeDefined();
    });

    it('it should render a toolbar', () => {
        const tree = shallow(
            <Toolbar />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });
});