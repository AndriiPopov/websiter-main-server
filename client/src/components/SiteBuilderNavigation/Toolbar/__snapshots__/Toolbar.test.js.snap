// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar it should render a toolbar 1`] = `
<header
  className="Toolbar"
>
  <nav
    className="DesktopOnly"
  >
    <NavigationItems />
  </nav>
  <div
    className="ToolSetsContainer"
  >
    <Connect(ToolsSet) />
  </div>
</header>
`;
